airbyte-commons/src/main/java/io/airbyte/commons/util/MoreLists.java

```diff
@@ -39,4 +41,18 @@
     return Optional.ofNullable(list.get(list.size() - 1));
   }
 
+  /**
+   * Reverses a list by creating a new list with the same elements of the input list and then
+   * reversing it. The input list will not be altered.
+   *
+   * @param list to reverse
+   * @param <T> type
+   * @return new list with elements of original reversed.
+   */
+  public static <T> List<T> reverse(List<T> list) {
```

Comment:
suggest renaming to `reversed` since it implies the copy

Reply:
cool